src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs

```diff
@@ -28,14 +28,12 @@ public BuildPublishTests(ITestOutputHelper output, SharedBuildPerTestClassFixtur
     [InlineData("Release")]
     public async Task DefaultTemplate_WithoutWorkload(string config)
     {
-        string id = $"blz_no_workload_{config}_{GetRandomId()}_{s_unicodeChars}";
-        CreateBlazorWasmTemplateProject(id);
+        ProjectInfo info = CopyTestAsset(config, aot: false, "BlazorBasicTestApp", "blz_no_workload", "App");
```

Comment:
idea: Could we replace the string for testasset with some static collection of possible assets and include the "App" in there? Sometime like:

```C#
public class TestAsset
{
    public string Name { get; init; }
    public string RunnableProjectSubPath { get; init; }
    
    public static readonly TestAsset BlazorBasicTestApp = new() { Name = "BlazorBasicTestApp", RunnableProjectSubPath = "App" };
}

...

CopyTestAsset(config, aot: false, TestAsset.BlazorBasicTestApp, "blz_no_workload");
```

Reply:
Good idea but I would prefer not to keep that knowledge in `TestAsset`. If it's `BlazorBasicTestApp` then it can become `BasicTestApp` and be kept by `BlazorWasmTestBase` and same with Wasm app.

Reply:
My thinking is that path to the application project is specific to testapp. If we would have two blazor testapps, both with different structure, we would have to encode that somewhere. I don't think base class is the right place, as I would like avoid having base class tight to a specific testapp (in general I much more prefer composition over inheritance, as it gives better flexibility to compose things together).

I think encoding testapp options/structure/parameters close to testapp is benefitial in terms of
- If one wants to use the testapp, he gets all the data.
- If one wants to add new testapp, he sees what needs to be set.

Maybe this comes also to the discussion about `TestScenario`, if it would be part of RunOptions specific for `WasmBasicTestApp`, it would be obvious that it should be used with it and nowhere else.

This might come to big change in the structure, so I'm just trying to share my thoughts about the structure how I see it usable